inbc-program/tests/unit/test_ota_parser.py

```diff
@@ -235,7 +235,21 @@ def test_create_sota_manifest(self, mock_pass, mock_reconnect) -> None:
         expected = '<?xml version="1.0" encoding="utf-8"?><manifest><type>ota</type><ota><header><type>sota</type' \
                    '><repo>remote</repo></header><type><sota><cmd ' \
                    'logtofile="y">update</cmd><mode>full</mode><package_list></package_list>' \
-                   '<fetch>https://abc.com/test.tar</fetch><username>Frank</username><password>123abc</password>' \
+                   '<fetch>https://abc.com/test.tar</fetch><signature></signature><username>Frank</username><password>123abc</password>' \
+                   '<release_date>2026-12-31</release_date><deviceReboot>yes</deviceReboot>' \
+                   '</sota></type></ota></manifest>'
+        self.assertEqual(s.func(s), expected)
+
+    @patch('inbm_lib.mqttclient.mqtt.mqtt.Client.reconnect')
+    @patch('inbc.utility.getpass.getpass', return_value='123abc')
+    def test_create_sota_manifest_with_signature(self, mock_pass, mock_reconnect) -> None:
+        s = self.arg_parser.parse_args(
+            ['sota', '-u', 'https://abc.com/test.tar', '-un', 'Frank', '-s', 'https://abc.com/signature.file'])
+        expected = '<?xml version="1.0" encoding="utf-8"?><manifest><type>ota</type><ota><header><type>sota</type' \
+                   '><repo>remote</repo></header><type><sota><cmd ' \
+                   'logtofile="y">update</cmd><mode>full</mode><package_list></package_list>' \
+                   '<fetch>https://abc.com/test.tar</fetch><signature>https://abc.com/signature.file</signature>' \
```

Comment:
the name `signature` is a bit confusing--I was expecting it to be a literal signature. perhaps `signature_uri`?

Reply:
sorry, this is the old test case. The signature should be the SHA256 checksum value.

Reply:
test case updated